rollback rule.validate(), IsDatetimeIsoFormat removed

d-ganchar opened the pull request.

d-ganchar commented on flask_request_validator/rules.py:
```diff
@@ -159,42 +136,23 @@ def __init__(self, value: Union[int, float], include_boundary: bool = True) -> N
-        return value, errors
-
-
-class IsDatetimeIsoFormat(AbstractRule):
```
@LostInDarkMath is not a `Rule`. Rules works with converted values. It should be a new `Param`

LostInDarkMath commented:
Can you please explain why you have removed this? Why is `IsDateTimeIsoFormat` not a rule? I don't get the point here.

d-ganchar commented:
> Can you please explain why you have removed this? Why is `IsDateTimeIsoFormat` not a rule? I don't get the point here...
> 
> And I think it's a shame that you deleted the release. I've already updated two systems on it and now nothing works anymore :(

`Rule` doesn't convert types. Look at [v3.0.0](https://github.com/d-ganchar/flask_request_validator/pull/50/files). Changed `Python versions`, added `HEADER` request type and that's all. Look at [patch](https://github.com/d-ganchar/flask_request_validator/pull/51/commits/e7ba6157835d2405e5f5f70c15e8ff8b15ccc3f0) changes. `AbstractRule.validate()` returns 2 items. We cannot change an interface in such versions. It means that  `patch` is not compatible with `release`. People will have broken `CustomRules` after upgrade. 
Yes - it was terrible decision. I really apologize but `Small improvements` was merged less then day ago. So I think it was a bad but necessary decision because 3.0.1 was broken.

About your systems... You can create `DateTimeIsoParam` and implement [value_to_type](https://github.com/d-ganchar/flask_request_validator/blob/master/flask_request_validator/validator.py#L53) as quick fix. 
Hope you understand ❤️

LostInDarkMath commented:
Yes, there was an breaking change in Version 3.0.0: `AbstractRule.validate()` returned a tuple of a value and a list of errors. In Version 2.1.2 and before, `AbstractRule.validate()` returned just a list of errors. So Version 3.0.0 wasn't compatible to version 2.1.2. That's why it is version 3.0.0 i thought. 

But Version 3.0.1 was fully downward compatible to version 3.0.0.

Not only you undo the changes in version 3.0.1, you also revert the changes of version 3.0.0 completely. Therefore, the old version 3.0.0 you released 22 days ago is completely incompatible to this "new" version 3.0.0 and will be completely incompatible to all upcoming versions simply because the interface was `AbstractRule.validate()`.

Because 22 days are a relative long period of time and you have ~350 downloads per month you broke maybe about 250 systems, not just mine. I mean shure, other systems may have not seen the release 3.0.1 and not updated yet, but if you release a new version, all those systems will break.

In a nutshell, I think you oversee here, that the breaking change was actually made in version 3.0.0 not in the yesterday published version 3.0.1.

d-ganchar commented:
🤦 🤕 🤒 I need to vacations... 
Thank you! 3.0.1 was uploaded to pypi.
